lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1957,6 +1957,99 @@ def plot(self, xs, ys, *args, zdir='z', **kwargs):
 
     plot3D = plot
 
+    def fill_between(self, x1, y1, z1, x2, y2, z2, *,
+                     where=None, single_polygon=False, **kwargs):
+        """
+        Fill the area between two 3D curves.
+
+        The curves are defined by the points (*x1*, *y1*, *z1*) and
+        (*x2*, *y2*, *z2*). This creates one or multiple quadrangle
+        polygons that are filled. All points must be the same length N, or a
+        single value to be used for all points.
+
+        Parameters
+        ----------
+        x1, y1, z1 : float or 1D array-like
+            x, y, and z  coordinates of vertices for 1st line.
+
+        x2, y2, z2 : float or 1D array-like
+            x, y, and z coordinates of vertices for 2nd line.
+
+        where : array of bool (length N), optional
+            Define *where* to exclude some regions from being filled. The
+            filled regions are defined by the coordinates ``pts[where]``,
+            for all x, y, and z pts. More precisely, fill between ``pts[i]``
+            and ``pts[i+1]`` if ``where[i] and where[i+1]``. Note that this
+            definition implies that an isolated *True* value between two
+            *False* values in *where* will not result in filling. Both sides of
+            the *True* position remain unfilled due to the adjacent *False*
+            values.
+
+        single_polygon : bool
```

Comment:
I think this is a brand new keyword argument and something similar isn't done in any of the other 3D areas? I'm personally not convinced of it and I think it should just be we draw a single polygon and then whatever happens downstream due to our 3d->2d rendering limitations is just that and can be improved later if we get a better renderer. (I also think it should be default True if we do add this as that is what I'd naively expect)

For example, if you do move the image around in your example:
```python
import matplotlib.pyplot as plt
import numpy as np

N = 50
theta = np.linspace(0, 2*np.pi, N)

x1 = np.cos(theta)
y1 = np.sin(theta)
z1 = 0.1 * np.sin(6 * theta)

x2 = 0.6 * np.cos(theta)
y2 = 0.6 * np.sin(theta)
z2 = 2  # Note that scalar values work in addition to length N arrays

fig = plt.figure()
ax = fig.add_subplot(projection='3d')
ax.fill_between(x1, y1, z1, x2, y2, z2, alpha=0.5, edgecolor='k', single_polygon=True)

plt.show()
```
You can see that the `single_polygon` case does look good from the right angle.

Reply:
It is correct, that this is a new concept, which is inherent to the task. There is no single canonical solution to create a surface between two arbitrary lines. It's thus reasonable to have multiple implementations. However, a bool flag is not good. It it selects or deselects a variant. But does not communicate what is used in the latter case instead. Also there can be other implementations, e.g. triangles instead of quadrangles. Let's use `mode` (or a similar name). I have no clear preference on the default. We could now or later even add a `mode="auto"` that tries to detect the best mode based on the input data.

Reply:
The single polygon case for that example does not render as intended, it looks like this:
![image](https://github.com/matplotlib/matplotlib/assets/14363975/b08ec27d-f5f5-4e34-9992-594fe94dd3ca)

From other angles it looks as intended, but this is actually a manifestation of https://github.com/matplotlib/matplotlib/issues/28394 and not correct.
![image](https://github.com/matplotlib/matplotlib/assets/14363975/df28842b-f096-445d-86b0-cca6ec563655)

Reply:
I like @timhoffm's suggestion to make this a `mode` selection rather than a flag. Proposed modes are `'auto', 'quad', 'polygon'` (better names suggested). For now I think `'auto'` should map to `'quad'`, since that is a superset of the polygon behavior and gives more flexible results.

Reply:
See also the single polygon mode not performing as intended on the double helix example: 
![image](https://github.com/matplotlib/matplotlib/assets/14363975/8d424d2a-ffe9-471f-a9cb-29a90e22a94f)

Reply:
I think I was getting hung up on wanting a known mapping of input values -> polygons so that the collection could be mapped in a known way by the `Collection`/`ScalarMappable`. That is already not the case though because we have the `where` argument and can get any number of `n` polygons being returned based on where the condition holds. So mapping already doesn't really make sense in this case.

I'm satisfied with having both options for constructing the "between" area.